troubleshooting/known_issues/kafka_chart_upgrade_failure.md

```diff
@@ -0,0 +1,151 @@
+```
+
+There would be failed Kubernetes jobs for Kafka as follows -
+
+```text
+ncn-m001:~/cray-kafka-operator # kubectl get job -A|grep kafka
+operators   cray-kafka-pre-upgrade-crd-hook   0/1   9m   9m
+
+ncn-m001:~/cray-kafka-operator # kubectl get po -n operators
+NAME                                    READY  STATUS   RESTARTS   AGE
+cray-kafka-pre-upgrade-crd-hook-7dtcx   0/1    Error    0          9m
+```
+
+The `/root/output.log` which stores the upgrade logs contains following type of errors related to Strimzi Kafka Custom Resource Definitions(CRDs) -
+
+```text
+Error from server (Invalid): error when applying patch:
+{"metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":<last_applied_configuration_values>
+to:
+Resource: "apiextensions.k8s.io/v1, Resource=customresourcedefinitions", GroupVersionKind: "apiextensions.k8s.io/v1, Kind=CustomResourceDefinition"
+Name: "kafkas.kafka.strimzi.io", Namespace: ""
+for: "/tmp/strimzi-crds-0.41.0.yaml": CustomResourceDefinition.apiextensions.k8s.io "kafkas.kafka.strimzi.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta1": must appear in spec.versions
+
+Error from server (Invalid): error when applying patch:
+{"metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":<last_applied_configuration_values>
+to:
+Resource: "apiextensions.k8s.io/v1, Resource=customresourcedefinitions", GroupVersionKind: "apiextensions.k8s.io/v1, Kind=CustomResourceDefinition"
+Name: "kafkaconnects.kafka.strimzi.io", Namespace: ""
+for: "/tmp/strimzi-crds-0.41.0.yaml": CustomResourceDefinition.apiextensions.k8s.io "kafkaconnects.kafka.strimzi.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta1": must appear in spec.versions
+
+Error from server (Invalid): error when applying patch:
+{"metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":<last_applied_configuration_values>
+to:
+Resource: "apiextensions.k8s.io/v1, Resource=customresourcedefinitions", GroupVersionKind: "apiextensions.k8s.io/v1, Kind=CustomResourceDefinition"
+Name: "kafkarebalances.kafka.strimzi.io", Namespace: ""
+for: "/tmp/strimzi-crds-0.41.0.yaml": CustomResourceDefinition.apiextensions.k8s.io "kafkarebalances.kafka.strimzi.io" is invalid: status.storedVersions[1]: Invalid value: "v1alpha1": must appear in spec.versions
+```
+
+Similar type of errors could be seen on other Strimzi CRDs too.
+
+## [Workaround steps](#workaround-steps)
+
+(`ncn-mw#`) To resolve this issue, update the Strimzi CRDs on the cluster by removing references to the older versions - `v1alpha1` and `v1beta1`.
+After this update, the Strimzi CRDs (except `kafkatopics` and `kafkausers`) will retain only the latest supported version `v1beta2` which eliminates the upgrade failure.
+Run the following script to fix the issue -
+
+[`kafka_crd_fix.sh`](../scripts/kafka_crd_fix.sh)
```

Comment:
This should give the full path to the script, so they can cut-n-paste it:
```
    ```bash
    /usr/share/doc/csm/troubleshooting/scripts/kafka_crd_fix.sh
    ```